six.py

```diff
@@ -843,6 +843,63 @@ def wrapper(cls):
     return wrapper
 
 
+def ensure_binary(s, encoding='utf-8', errors='strict'):
+    """ A helper function to ensure output is six.binary_type.
+
+    For Python 2:
+      - `unicode` -> encoded to `str`
+      - `str` -> `str`
+
+    For Python 3:
+      - `str` -> encoded to `bytes`
+      - `bytes` -> `bytes`
+    """
+    if not isinstance(s, (text_type, binary_type)):
+        raise TypeError("not expecting type '%s'" % type(s))
+    if isinstance(s, text_type):
+        return s.encode(encoding, errors)
+    else:
```

Comment:
@benjaminp do you mean something like?
```
def ensure_binary(s, encoding='utf-8', errors='strict'):
    if isinstance(s, text_type):
        return s.encode(encoding, errors)
    elif isinstance(s, binary_type):
        return s
    else:
        raise TypeError("not expecting type '%s'" % type(s))
```
I'd like to keep the `raise TypeError` part to fail the error loudly; be consistent with the `raise` behavior from `ensure_string`.